### PR TITLE
Update usaco-892.mdx - added cpp solution

### PR DESCRIPTION
**Pull request by sathu-beep**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: solutions/usaco-892.mdx [hunk 1]
```diff
@@ -74,4 +74,35 @@ public class SleepyCowSorting {
...
+int main() {
+    	freopen("sleepy.in", "r", stdin);
+    	freopen("sleepy.out", "w", stdout);
+	int n; cin >> n;
+	for (int i = 0; i < n; i++) {
+		cin >> arr[i];
+	}
+	int answer = n - 1;
+	for (int i = n - 2; i >= 0; i--) {
+		if(arr[i] < arr[i+1]) {
```
Comment (brad182) [inline solutions/usaco-892.mdx:94]: other people don't want this if else style but I'm fine with it

File: solutions/usaco-892.mdx [hunk 2]
```diff
@@ -74,4 +74,35 @@ public class SleepyCowSorting {
...
+			answer = i;
+        	}
+		else {
+			break;
+        	}
+	}
+	cout << answer << endl;	
+}
+
```
Comment (brad182) [inline solutions/usaco-892.mdx:103]:
```suggestion
```

File: solutions/usaco-892.mdx [hunk 3]
```diff
@@ -74,4 +74,35 @@ public class SleepyCowSorting {
 	
 </JavaSection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int arr[105];
```
Comment (brad182) [inline solutions/usaco-892.mdx:83]: change name to something else that describes what it stores

**Review by brad182 — APPROVED** [review]

File: solutions/usaco-892.mdx (hunk 1 above)
Comment (sathu-beep) [reply solutions/usaco-892.mdx]: uh what do you mean

File: solutions/usaco-892.mdx (hunk 3 above)
Comment (sathu-beep) [reply solutions/usaco-892.mdx]: fixed :D

File: solutions/usaco-892.mdx [hunk 4]
```diff
@@ -74,4 +74,34 @@ public class SleepyCowSorting {
...
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int cows[105];
+
+int main() {
+    	freopen("sleepy.in", "r", stdin);
+    	freopen("sleepy.out", "w", stdout);
```
Comment (dongliu0426) [inline solutions/usaco-892.mdx:87]:
```suggestion
	freopen("sleepy.in", "r", stdin);
	freopen("sleepy.out", "w", stdout);
```

**Pull request merged**